beets/util/queue.py

```diff
@@ -146,10 +153,9 @@ def put(self, item):
         return immediately.
         """
         with self.not_full:
-            if not self._invalidated and self._maxsize > 0:
-                while (not self._invalidated and
-                       len(self._queue) >= self._maxsize):
-                    self.not_full.wait()
+            while (not self._invalidated and self._is_full()):
```

Comment:
This collapse of the `if` and `while` looks right to me with the above change to `_is_full`. But I think it's incorrect without that change: the intended behavior is to skip waiting altogether if `_maxsize == 0`, and I don't think we'd do that with the current version.

Reply:
Yes, `_is_full` was supposed to do what its name suggests.